File: src/test/java/br/com/model/entities/classes/usuario/FornecedorTest.java
package br.com.model.entities.classes.usuario;

import org.junit.jupiter.api.Test;

import br.com.newObjects.model.entities.classes.NovoProduto;
import br.com.newObjects.model.entities.classes.NovoProdutoFornecedor;
import br.com.newObjects.model.entities.classes.NovoTelefone;
import br.com.newObjects.model.entities.classes.usuario.NovoFornecedor;

public class FornecedorTest {
    @Test
    void testAdicionarProduto() {
        var a = NovoFornecedor.novoFornecedor();
        a.adicionarProduto(NovoProdutoFornecedor.novoProdutoFornecedor());
        System.out.println(a.toString());
    }

    @Test
    void testAdicionarTelefone() {
        var a = NovoFornecedor.novoFornecedor();
        a.adicionarTelefone(NovoTelefone.novoTelefone(a));
        System.out.println(a.toString());
    }

    @Test
    void testAtualizarStatusPedido() {
        // TODO
        // NovoFornecedor.novoFornecedor().atualizarStatusPedido();
    }

    @Test
    void testBuscarFornecedor() {
        // TODO
    }

    @Test
    void testConfirmarEntrega() {
        // TODO
        // só dá pra fazer isso se o fornecedor tiver um pedido.
        // NovoFornecedor.novoFornecedor().confirmarEntrega();
    }

    @Test
    void testConfirmarEstoque() {
        var a = NovoFornecedor.novoFornecedor();
        a.confirmarEstoque(100, NovoProduto.novoProduto());
        System.out.println(a.toString());
    }

    @Test
    void testConfirmarPedido() {
        // TODO
        // Só dá pra fazer isso se o fornecedor tiver um pedido.
        // NovoFornecedor.novoFornecedor().confirmarPedido();
    }
}

File: src/test/java/br/com/model/FornecedorTest.kt
package br.com.model

import br.com.newObjects.model.NovoProduto
import br.com.newObjects.model.NovoProdutoFornecedor
import br.com.newObjects.model.NovoTelefone.novoTelefone
import br.com.newObjects.model.usuario.NovoFornecedor
import org.junit.jupiter.api.Test

class FornecedorTest {
    @Test
    fun testAdicionarProduto() {
        val a = NovoFornecedor.novoFornecedor()
        a.adicionarProduto(NovoProdutoFornecedor.novoProdutoFornecedor())
        println(a.toString())
    }

    @Test
    fun testAdicionarTelefone() {
        val a = NovoFornecedor.novoFornecedor()
        novoTelefone(a)
        println(a.toString())
    }

    @Test
    fun testAtualizarStatusPedido() {
        // TODO
        // NovoFornecedor.novoFornecedor().atualizarStatusPedido();
    }

    @Test
    fun testBuscarFornecedor() {
        // TODO
    }

    @Test
    fun testConfirmarEntrega() {
        // TODO
        // só dá pra fazer isso se o fornecedor tiver um pedido.
        // NovoFornecedor.novoFornecedor().confirmarEntrega();
    }

    @Test
    fun testConfirmarEstoque() {
        val a = NovoFornecedor.novoFornecedor()
        a.confirmarEstoque(100, NovoProduto.novoProduto())
        println(a.toString())
    }

    @Test
    fun testConfirmarPedido() {
        // TODO
        // Só dá pra fazer isso se o fornecedor tiver um pedido.
        // NovoFornecedor.novoFornecedor().confirmarPedido();
    }
}
